src/libraries/System.Runtime.Extensions/tests/System/EnvironmentTests.cs

```diff
@@ -154,6 +154,18 @@ public void OSVersion_ParseVersion(string input, int major, int minor, int build
             Assert.Equal(expected, actual);
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        public void OSVersion_ValidVersion_OSX()
+        {
+            Version version = Environment.OSVersion.Version;
+
+            // verify that the Environment.OSVersion.Version matches the current RID
+            Assert.Contains(version.ToString(2), RuntimeInformation.RuntimeIdentifier);
+
+            Assert.Equal(-1, version.Revision); // Revision is never set on OSX
```

Comment:
It looks like we're testing the 1st, 2nd, and 4th parts. Anything we can assert about the 3rd?

Reply:
When I first thought about it nothing came to mind. Thinking a little again, the one thing that does come to mind is that it is non-negative. I wouldn't think OSX would ever use a negative patch number.